# ansible-playbook -i hosts cs-svc.yml
---
- name: up and running docker containers
  hosts: all
  become: yes
  gather_facts: yes
  vars_files:
  - ./config/default.yml

  tasks:
  - name: "Wait for {{ default_transport }}://{{ certbot_domain_name }} service is up"
    uri:
      url: "{{ default_transport }}://{{ certbot_domain_name }}"
      method: GET
      headers:
      return_content: yes
    register: _result
    until: _result.content != '\"\"'
    retries: 30   # retry X times
    delay: 3      # pause for X sec b/w each call

  - name: "Check PKC MediaWiki Service - {{ default_transport }}://{{ certbot_domain_name }}"
    uri: "url={{ default_transport }}://{{ certbot_domain_name }} return_content=true"
    register: response
    failed_when: "'Main Page' not in response.content"

  # - debug:
  #     msg: "{{ response }}"

  - name: "Check Gitea Service - {{ default_transport }}://git.{{ certbot_domain_name }}"
    uri: "url={{ default_transport }}://git.{{ certbot_domain_name }} return_content=true"
    register: response
    failed_when: "'Sign In' not in response.content"

  # - debug:
  #     msg: "{{ response }}"

  - name: "Check Matomo Service - {{ default_transport }}://mtm.{{ certbot_domain_name }}"
    uri: "url={{ default_transport }}://mtm.{{ certbot_domain_name }} return_content=true"
    register: response
    failed_when: "'Sign in' not in response.content"

  # - debug:
  #     msg: "{{ response }}"

  - name: "Check PHP My Admin Service - {{ default_transport }}://pma.{{ certbot_domain_name }}"
    uri: "url={{ default_transport }}://pma.{{ certbot_domain_name }} return_content=true"
    register: response
    failed_when: "'Log in' not in response.content"

  # - debug:
  #     msg: "{{ response }}"

  - name: "Check Code Server Service - {{ default_transport }}://code.{{ certbot_domain_name }}"
    uri:
      url: "{{ default_transport }}://code.{{ certbot_domain_name }}"
      status_code: 200

  # - debug:
  #     msg: "{{ response }}"



